filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala

```diff
@@ -44,55 +46,71 @@ object GcsPathBuilder {
   }
 
   def getUri(string: String) = URI.create(UrlEscapers.urlFragmentEscaper().escape(string))
-}
+  
+  def fromAuthMode(authMode: GoogleAuthMode,
+                   applicationName: String,
+                   retrySettings: Option[RetrySettings],
+                   cloudStorageConfiguration: CloudStorageConfiguration,
+                   options: WorkflowOptions)(implicit as: ActorSystem, ec: ExecutionContext): Future[GcsPathBuilder] = {
+    authMode.credential(options) map { credentials =>
+      fromCredentials(credentials,
+        applicationName,
+        retrySettings,
+        cloudStorageConfiguration,
+        options
+      )
+    }
+  }
+  
+  def fromCredentials(credentials: Credentials,
+                      applicationName: String,
+                      retrySettings: Option[RetrySettings],
+                      cloudStorageConfiguration: CloudStorageConfiguration,
+                      options: WorkflowOptions): GcsPathBuilder = {
+    val transportOptions = HttpTransportOptions.newBuilder()
+      .setReadTimeout(3.minutes.toMillis.toInt)
+      .build()
 
-class GcsPathBuilder(authMode: GoogleAuthMode,
-                     applicationName: String,
-                     retrySettings: Option[RetrySettings],
-                     cloudStorageConfiguration: CloudStorageConfiguration,
-                     options: WorkflowOptions) extends PathBuilder {
-  authMode.validate(options)
+    val storageOptionsBuilder = StorageOptions.newBuilder()
+      .setTransportOptions(transportOptions)
+      .setCredentials(credentials)
 
-  protected val transportOptions = HttpTransportOptions.newBuilder()
-    .setReadTimeout(3.minutes.toMillis.toInt)
-    .build()
+    retrySettings foreach storageOptionsBuilder.setRetrySettings
```

Comment:
are you sure the builder is mutable?  I.e. this might return the modified builder